pkg/genericcontrolplane/server.go

```diff
@@ -97,25 +93,43 @@ func Run(completeOptions completedServerRunOptions, stopCh <-chan struct{}) erro
 	// To help debugging, immediately log version
 	klog.Infof("Version: %+v", version.Get())
 
-	server, err := CreateServerChain(completeOptions, stopCh)
+	serverChain, err := CreateServerChain(completeOptions, stopCh)
 	if err != nil {
 		return err
 	}
+	server := serverChain.MiniAggregator.GenericAPIServer
 
 	prepared := server.PrepareRun()
 	return prepared.Run(stopCh)
 }
 
+type ServerChain struct {
+	CustomResourceDefinitions *apiextensionsapiserver.CustomResourceDefinitions
+	GenericControlPlane       *apis.GenericControlPlane
+	MiniAggregator            *MiniAggregatorServer
+}
+
 // CreateServerChain creates the apiservers connected via delegation.
-func CreateServerChain(completedOptions completedServerRunOptions, stopCh <-chan struct{}) (*genericapiserver.GenericAPIServer, error) {
+func CreateServerChain(completedOptions completedServerRunOptions, stopCh <-chan struct{}) (*ServerChain, error) {
```

Comment:
This change may not be ideal, but I definitely make use of CustomResourceDefinitions and GenericControlPlane in the kcp server code.